launcher/launcher.go

```diff
@@ -101,48 +98,84 @@ func (l *launcher) Run(b *beat.Beat) error {
 }
 
 func (l *launcher) run() error {
-	l.log.Info("Beater launcher is running")
-	err := l.runBeater()
-	if err != nil {
-		l.log.Errorf("Could not run Beater: %v", err)
-		return err
-	}
-
-	err = l.waitForUpdates()
+	err := l.runLoop()
 	if err != nil {
-		l.log.Errorf("Beater launcher is stopping: %v", err)
+		l.log.Errorf("Launcher has stopped: %v", err)
 	} else {
-		l.log.Info("Beater launcher was shutted down gracefully")
+		l.log.Info("Launcher was shutted down gracefully")
 	}
+
+	l.reloader.Stop()
 	return err
 }
 
+// runLoop is the loop that keeps the launcher alive
+func (l *launcher) runLoop() error {
+	l.log.Info("Launcher is running")
+	for {
+		// Run a new beater
+		err := l.runBeater()
+		if err != nil {
+			return fmt.Errorf("launcher could not run Beater: %v", err)
+		}
+
+		// Wait for something to happen:
+		// config update produces val, nil
+		// signal produces nil, nil
+		// beater error produces nil, err
+		cfg, err := l.waitForUpdates()
+
+		// If it's not a beater error, should stop the beater
```

Comment:
Are we sure we don't want to stop the beater in case of an error?
Are there any consequences?

Reply:
In case it's a beater error the beater is no longer running.